tools/marvin/marvin/lib/base.py

```diff
@@ -3377,6 +3377,40 @@ def list(cls, apiclient, **kwargs):
             cmd.listall = True
         return(apiclient.listPortableIpRanges(cmd))
 
+
+class NuageUnderlayPublicIpRange:
```

Comment:
I would avoid having non-core components/base resource classes in base.py/marvin. Instead such a class can be moved into a utility module at test/integration/plugins/nuagevsp/